### PR TITLE
Report ignorefile location when no content added

### PR DESCRIPTION
**Pull request by rhatdan**
Users have accidently had a .containerignore or .dockerignore
file in their context directly which blocked all content.
Currently we report that no globs matched, but do not
identify where the globs came from.

This change is an attempt to add this data to the error.
Example: https://github.com/containers/buildah/issues/3318

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: add.go [hunk 1]
```diff
@@ -564,7 +564,17 @@ func (b *Builder) Add(destination string, extract bool, options AddAndCopyOption
...
 		}
 		if itemsCopied == 0 {
-			return errors.Wrapf(syscall.ENOENT, "no items matching glob %q copied (%d filtered out)", localSourceStat.Glob, len(localSourceStat.Globbed))
+			ignoreFile := filepath.Join(contextDir, ".containerignore")
+			if _, err := os.Stat(ignoreFile); err != nil {
+				ignoreFile = filepath.Join(contextDir, ".dockerignore")
+				if _, err := os.Stat(ignoreFile); err != nil {
+					ignoreFile = ""
+				}
+			}
```
Comment (nalind) [inline add.go:573]: This guess could well be wrong if `buildah add`'s `--ignorefile` flag is used.

Comment (rhatdan) [reply add.go:573]: There is no way to know at this level whether that happened. I would figure most likely, their would not have been an ignorefile.

File: pkg/parse/parse.go [hunk 2]
```diff
@@ -1098,3 +1100,47 @@ func SSH(sshSources []string) (map[string]*sshagent.Source, error) {
...
+func ignoreFile(contextDir string) string {
+	for _, i := range []string{".containerignore", ".dockerignore"} {
+		ignoreFile := filepath.Join(contextDir, i)
+		if _, err := os.Stat(ignoreFile); err == nil {
+			return ignoreFile
+		}
+	}
+	return ""
+}
+func ContainerIgnoreFile(path string, contextDir string) ([]string, string, error) {
```
Comment (nalind) [inline pkg/parse/parse.go:1128]: The arguments here are in the reverse of the order in which I think its callers expect.

File: pkg/parse/parse.go [hunk 3]
```diff
@@ -1098,3 +1100,47 @@ func SSH(sshSources []string) (map[string]*sshagent.Source, error) {
...
+	for _, i := range []string{".containerignore", ".dockerignore"} {
+		ignoreFile := filepath.Join(contextDir, i)
+		if _, err := os.Stat(ignoreFile); err == nil {
+			return ignoreFile
+		}
+	}
+	return ""
+}
+func ContainerIgnoreFile(contextDir, path string) ([]string, string, error) {
+	if path != "" {
```
Comment (nalind) [inline pkg/parse/parse.go:1129]: Might as well use `imagebuilder.ParseIgnore()` instead of most of this block.

**Comment by rhatdan** [comment]
@nalind @TomSweeneyRedHat @flouthoc @vrothberg @giuseppe @ashley-cui @umohnani8 PTAL

**Pull request merged**